src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -310,23 +311,25 @@ def _check_registry_health(cmd, registry_name, ignore_errors):
     if status_validated:
         _get_endpoint_and_token_status(cmd, login_server, ignore_errors)
 
-    # CMK settings
-    if registry and registry.encryption and registry.encryption.key_vault_properties:  # pylint: disable=too-many-nested-blocks
-        client_id = registry.encryption.key_vault_properties.identity
-        valid_identity = False
-        if registry.identity:
-            valid_identity = (client_id == 'system') and bool(registry.identity.principal_id)  # use system identity?
-            if not valid_identity and registry.identity.user_assigned_identities:
-                for k, v in registry.identity.user_assigned_identities.items():
-                    if v.client_id == client_id:
-                        from msrestazure.azure_exceptions import CloudError
-                        try:
-                            valid_identity = (resolve_identity_client_id(cmd.cli_ctx, k) == client_id)
-                        except CloudError:
-                            pass
-        if not valid_identity:
-            from ._errors import CMK_MANAGED_IDENTITY_ERROR
-            _handle_error(CMK_MANAGED_IDENTITY_ERROR.format_error_message(registry_name), ignore_errors)
+    if cmd.supported_api_version(min_api='2020-11-01-preview', resource_type=ResourceType.MGMT_CONTAINERREGISTRY):  # pylint: disable=too-many-nested-blocks
+        # CMK settings
```

Comment:
please also add acr code owner to review your change

Reply:
@Bhuvaneswari-Santharam I tested this locally and it looks good. @northtyphoon we should consider adding some unit tests for `acr check-health`